app/src/main/java/org/mozilla/fenix/HomeActivity.kt

```diff
@@ -723,5 +718,23 @@ open class HomeActivity : LocaleAwareAppCompatActivity(), NavHostActivity {
         // PWA must have been used within last 30 days to be considered "recently used" for the
         // telemetry purposes.
         const val PWA_RECENTLY_USED_THRESHOLD = DateUtils.DAY_IN_MILLIS * 30L
+
+        /**
+         * set a delay until we locate visual completeness queue.
+         * static function to avoid memory leak in case we are trying to access visualCompletenessQueue
+         * or other activity members when activity is already destroy.
+         */
+        private fun setupRootContainerPreDrawDelay(
+            rootContainer: View,
+            homeActivityWeakRef: WeakReference<HomeActivity>
+        ) {
+            rootContainer.doOnPreDraw {
+                // This delay is temporary. We are delaying 5 seconds until the performance
+                // team can locate the real point of visual completeness.
+                it.postDelayed({
+                    homeActivityWeakRef.get()?.visualCompletenessQueue!!.ready()
```

Comment:
Why do we need a WeakReference to HomeActivity? Could we just pass in the visualCompletenessQUeue instead? I noticed it's already part of `Components`: `components.performance.visualCompletenessQueue`. So it shouldn't be dependent on the Activity.